<?xml version="1.0"?>


<launch>
<!-- Includes -->



  <!-- start nodelet manager -->
  <!--node pkg="nodelet" type="nodelet" name="pcl_manager"  args="manager" output="screen"/-->

  <!-- Start point cloud converter -->
	<!--node pkg="point_cloud_converter" type="point_cloud_converter" name="converter_mug" output="screen">
 	 	<remap from="/points_in" to="/full_cloud" />
 	 	<remap from="/points2_out" to="/converter_out" />
	</node-->
  	
  <node pkg="cob_env_model" type="field_of_view_node" name="field_of_view_node" output="screen"/>

	<!-- Start aggregate point map -->
	<node pkg="nodelet" type="nodelet" name="aggregate_point_map" args="load cob_env_model/AggregatePointMap cam3d_nodelet_manager2" output="screen">
  	<remap from="point_cloud2" to="/cam3d/depth_registered/points"/>
  	<remap from="point_cloud2_map" to="/aggregate_point_map/map"/>
  	<remap from="point_cloud2_aligned" to="/aggregate_point_map/pc_aligned"/>
  	
  	<param name="icp_max_iterations" type="int" value="50"/>
  	<param name="icp_max_corr_dist" type="double" value="0.2"/>
  	<param name="icp_trf_epsilon" type="double" value="5e-6"/>
  	<param name="file_path" type="string" value="/media/GOADaten/Daten/20110823_bagfiles/kitchen_test/icp/"/>
  	<param name="save_pc" type="bool" value="false"/>
  	<param name="save_map" type="bool" value="true"/>
  	<param name="save_pc_aligned" type="bool" value="true"/>
  	<param name="save_map_fov" type="bool" value="false"/>
  	<param name="save_pc_trans" type="bool" value="true"/>
  	<param name="voxel_leafsize_x" type="double" value="0.03" />
  	<param name="voxel_leafsize_y" type="double" value="0.03" />
  	<param name="voxel_leafsize_z" type="double" value="0.03" />
  	<param name="r_limit_" type="double" value="0.03" />
  	<param name="p_limit_" type="double" value="0.03" />
  	<param name="y_limit" type="double" value="0.03" />
  	<param name="distance_limit_" type="double" value="0.05"/>
  	<param name="use_reference_map" type="bool" value="true"/>
  </node>  
  
  </launch>
